### PR TITLE
Allow us to precompile Regexes at script compile time.

### PR DESCRIPTION
**Pull request by AWoloszyn**

### CODE REVIEW

**Comment by devpow112** [comment]
Issues with formatting
```
Formatting ./Libraries/WNContainers/inc/WNDynamicArray.h
./Libraries/WNContainers/inc/WNDynamicArray.h:684:37: error: code should be clang-formatted [-Wclang-format-violations]
      const size_type new_capacity = 
                                    ^
...
Formatting ./Libraries/WNContainers/inc/WNContiguousRange.h
./Libraries/WNContainers/inc/WNContiguousRange.h:47:45: error: code should be clang-formatted [-Wclang-format-violations]
  WN_FORCE_INLINE contiguous_range_iterator(contiguous_range_iterator&& _other) noexcept
```

File: Libraries/WNContainers/inc/optional.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+#ifndef __WN_CONTAINERS_OPTIONAL_H__
+#define __WN_CONTAINERS_OPTIONAL_H__
+
+#include <optional>
+
+namespace wn {
+namespace containers {
+
+using std::optional;
```
Comment (devpow112) [inline Libraries/WNContainers/inc/optional.h:15]: Would this be more of a core thing? It's not really a container

Comment (AWoloszyn) [reply Libraries/WNContainers/inc/optional.h:15]: Yeah we could put it in core. Will do right now

**Review by devpow112 — APPROVED** [review]

**Pull request merged**